### PR TITLE
Complete: Task 673-678: Ohsumed & Onestop_english Dataset

### PR DESCRIPTION
**Pull request by SaicharanPapani**
Used the Onestop_english and Ohsumed dataset and created the Tasks 673-677.

Please review and let me know me the corrections if any.

### CODE REVIEW

**Comment by swarooprm** [comment]
@kuntalkumarpal, wondering if you can review this PR?

**Comment by kuntalkumarpal** [comment]
In task674_ohsumed_medical_subject_headings_answer_generation.json, what does /* mean in the term "Cellulose/*" ?

**Comment by kuntalkumarpal** [comment]
For tasks/task675_ohsumed_yes_no_numerical_answer_generation.json

File: tasks/README.md [hunk 1]
```diff
@@ -6,4 +6,9 @@ This directory contains the tasks that are part of this benchmark.
 Name | Summary | Category
 ---- | ----------- | --------
 `task1001_squad1.1_question_generation` | Generating guestions (based on SQuAD 1.1) | Question Generation  
-`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
\ No newline at end of file
+`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
+`task673_ohsumed_abstract_title_generation` | Generating title to Ohsumed dataset abstracts | title Generation
```
Comment (kuntalkumarpal) [inline tasks/README.md:10]: Typo : Capitalize T in readme and data file as well for category

**Comment by kuntalkumarpal** [comment]
For tasks/task676_onestop_english_readability_classification.json
What is the definition of reading levels you are considering (0 - Elementary, 1 - Intermediate and 2 - Advance) ? is it only based on word count ?

**Comment by kuntalkumarpal** [comment]
For tasks/task677_ohsumed_yes_no_answer_generation.json,

- the definition is confusing "Does the following mesh (Medical subject heading) terms have a suitable mesh term for the abstract?".
- Is it the case that given the options, multiple options cannot be correct ?

File: tasks/README.md (hunk 1 above)
Comment (SaicharanPapani) [reply tasks/README.md]: It is done.

**Comment by SaicharanPapani** [comment]
1. Typo : Capitalize T in title for category both in readme and the generated file: **Fixed**
2. For tasks/task675_ohsumed_yes_no_numerical_answer_generation.json. Do you also consider numerical data in word for like "twelve" or "one" etc ? **- I haven't considered numbers in word format, but I have considered it now and fixed it.**
3. In task674_ohsumed_medical_subject_headings_answer_generation.json, what does '/\*' mean in the term "Cellulose/\*" ?  - **These Mesh terms are human assigned mesh terms based on the context and domain of the abstract. From what i understood, '/\*' means all the relevant terms that comes under Cellulose (Like regex, And there is huge set [maintained here](https://www.nlm.nih.gov/mesh/meshhome.html)). I think it better to remove such human understandable notation, so removing '/*'. Let me know, If there are any other changes.**
4. For tasks/task676_onestop_english_readability_classification.json. What is the definition of reading levels you are considering (0 - Elementary, 1 - Intermediate and 2 - Advance) ? is it only based on word count ? - **No, In the paper it is mentioned that it depends on word count, std deviation, POS tagging, ngrams etc. As It is not possible now for me to build a model, analyze it and get all these values, I generalized it using word count. let me know, If I should explain it in some other way (updated the definition).**
5. For tasks/task677_ohsumed_yes_no_answer_generation.json, the definition is confusing "Does the following mesh (Medical subject heading) terms have a suitable mesh term for the abstract?". Is it the case that given the options, multiple options cannot be correct ? - **No, it is like given a few MESH terms, And is there a suitable mesh term that is apt for the abstract?**

**Comment by SaicharanPapani** [comment]
I hope i had made the corrections. @kuntalkumarpal please review and let me know if there are any issues.

**Comment by kuntalkumarpal** [comment]
@swarooprm  Do you think in the definition of task 676 : "Classification depends on various factors like word count, standard deviation, POS tagging, ngrams etc." is correct or it should be reverted back to generic definition. Since this instruction may confuse the model.

**Comment by kuntalkumarpal** [comment]
@swarooprm looks good to me. Except definition of task 676

**Comment by SaicharanPapani** [comment]
@swarooprm Please let me know. If i should change that (suggest me a definition, if possible).

**Review by swarooprm — COMMENTED** [review]
Task676: Is it always true that task category is done purely based on word count? I infer the same from your explanation. Verify and if this is true, modify the definition to only consider word count. Otherwise modify "Classification depends on various factors like word count, standard deviation, POS tagging, ngrams etc." based on what factors it really depends and change explanation accordingly.

**Comment by SaicharanPapani** [comment]
@swarooprm 
The classification depends not only on word length but also on standard deviation. Advanced texts are much longer than elementary texts. However, the standard deviation for each level is also high, indicating that text length may not be the deciding factor in terms of reading level.

Updated the explanation as per the corpus (As given in the paper). 

They also performed the analysis using the following features:
Features | Accuracy
Word n-grams | 61.38%
POS n-grams | 67.37%
Char n-grams | 77.25%
Syntactic Production Rules | 54.67%
Dependency Relations | 27.16%

Let me know, If the updated explainations are fine or not.

**Comment by swarooprm** [comment]
Your HW2 is complete. 
There is still a problem with Task 676 explanation because in your explanation you mention standard deviation which can not be found for a single input, and rather it's an average of all inputs. I think this is how you can change your explanation: 
"The number of words in the text is 670 which is within the standard deviation is 117.15 of the average number of words in an intermediate level text (676.59). Based on its properties such as n-grams (word, POS and char-level) and syntactic production rules, it is an Intermediate level text and is classified as 1." 
Also, I would say mention all these features in the definition such as n-grams (word, POS, and char-level) and syntactic production rules.

Would you like to submit the same tasks to our big project repo https://github.com/allenai/natural-instructions-expansion/pulls? You just need to fork that repo, rename tasks starting from the latest number in the repo (which at this point is 442) and create PR there. This way, you can be part of our research on "learning from instructions instead of learning from examples (conventional machine learning)". This is purely optional and your HW2 grade does not depend on this @SaicharanPapani

**Comment by swarooprm** [comment]
Closing this assuming that you will update your files based on my above comment.

**Comment by SaicharanPapani** [comment]
> Closing this assuming that you will update your files based on my above comment.

Sure @swarooprm, I will note the changes for 676.
Should i copy paste the same tasks to big project repo https://github.com/allenai/natural-instructions-expansion/pulls?

**Comment by SaicharanPapani** [comment]
Can you open this PR? I will make the necessary changes and push them.

**Comment by swarooprm** [comment]
@SaicharanPapani, I have reopened the PR.
Please update task676 and then submit the same files as PR to https://github.com/allenai/natural-instructions-expansion/pulls? You just need to change filenames starting from the latest task number in repo (607)?

**Comment by SaicharanPapani** [comment]
@swarooprm 
I have updated the task 676 explainations. please verify and let me know. I will raise the PR for https://github.com/allenai/natural-instructions-expansion/pulls.

After verifying. If it seems good, please drop a comment. I will proceed for other PR to allenai.

**Comment by swarooprm** [comment]
@SaicharanPapani Looks good! Thanks.
Please go ahead and create PR in Allen AI repo.

**Pull request closed without merging**